### PR TITLE
Updated district chart to include a 'rate by exit' mode

### LINKED ISSUE #419: [Lantern] Update District chart to include a third "rate by exit" mode
**Is your feature request related to a problem? Please describe.**
As described in the parent epic #395, we want to add a third "mode" to the district chart in the Lantern view. We currently have "Revocation count" and "Percent revoked" selections -- "Percent revoked" is calculated as the total number of revocations divided by the total standing population, i.e. `revocationApiData.population_count / supervisionApiData.total_population`. The third mode will be a calculation of revocation rate but with a new denominator of _total exits_ from supervision.

**Describe the solution you'd like**
As noted in #395, add a third mode toggle to the District chart and update the text and styling for all three toggles to look like this:
image

When the third one ("Percent revoked of exits") is selected, it should function similarly to the "Percent revoked of standing population" mode (which is our _current_ "Percent revoked"), but should use a different denominator. That new denominator will be in the existing `revocations_matrix_supervision_distribution_by_district` file, which will be given a new column of `total_exit_count`. So we should use `total_exit_count` as the denominator when we're in this new mode, and continue to use `total_population` as the denominator in the existing "Percent revoked" mode.

**Describe alternatives you've considered**
N/A

**Additional context**
This is a child of the epic #395.

### LINKED ISSUE #420: [Lantern] Update risk level, gender, and race charts below the matrix to include a new "rate by exit" mode
**Is your feature request related to a problem? Please describe.**
As described in the parent epic #395, as a follow-up to #419, we also want to add a new revocation rate by exit mode to the risk level, gender, and race charts below the revocation matrix. #419 updated the district chart to have a new mode which calculations revocation rate with a new denominator. These three additional charts already display revocation rates calculated in the same way as the original mode in the district chart (i.e. the denominator is the total standing population

**Describe the solution you'd like**
As noted in #395, add two mode toggles to these three charts and update the text and styling for all three toggles to look like this:
image

(But without the first toggle in this case, since we don't want a general count mode for these three charts.)

When the "Percent revoked of standing population" mode is selected (the default), it should perform the calculations it currently performs today, with `population_count` as the numerator and `total_supervision_count` as the denominator.

When the "Percent revoked of exits" mode is selected, it should function similarly, but instead use a new denominator added to the existing chart files: `total_exit_count`. 

**Describe alternatives you've considered**
N/A

**Additional context**
This is a child of the epic #395. It should happen _after_ #419.

### PR DESCRIPTION
**Pull request by DmitryJakubovsky**
## Description of the change

* Changed Chip component
* Changed mode switcher in the district chart
* Split district chars to separated charts for more control

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #419
Closes #420 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  };
+
+  const chart = (
+    <Bar
+      id={chartId}
+      data={{
+        labels,
+        datasets: [
+          {
+            label: "Percent revoked of exits",
```
Comment (jessex) [inline src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js:95]: Let's call this "Percent revoked out of all exits"

File: src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+              },
+              stacked: true,
+            },
+          ],
+          yAxes: [
+            {
+              id: "y-axis-0",
+              scaleLabel: {
+                display: true,
+                labelString: "Percent revoked of exits",
```
Comment (jessex) [inline src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js:123]: Let's call this "Percent revoked out of all exits"

File: src/components/charts/new_revocations/RevocationsByDistrict/RatesChart.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  };
+
+  const chart = (
+    <Bar
+      id={chartId}
+      data={{
+        labels,
+        datasets: [
+          {
+            label: "Percent revoked",
```
Comment (jessex) [inline src/components/charts/new_revocations/RevocationsByDistrict/RatesChart.js:95]: Let's call this "Percent of standing population revoked"

File: src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js (hunk 2 above)
Comment (DmitryJakubovsky) [reply src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js:123]: Done

File: src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js (hunk 1 above)
Comment (DmitryJakubovsky) [reply src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js:95]: Done

File: src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
+// Recidiviz - a data platform for criminal justice reform
```
[the remark this answers is not in the document]
Comment (DmitryJakubovsky) [reply src/components/charts/new_revocations/RevocationsByDistrict/ExitesChart.js:1]: Done

File: src/components/charts/new_revocations/RevocationsByDistrict/RatesChart.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+              },
+              stacked: true,
+            },
+          ],
+          yAxes: [
+            {
+              id: "y-axis-0",
+              scaleLabel: {
+                display: true,
+                labelString: "Percent revoked",
```
[the remark this answers is not in the document]
Comment (DmitryJakubovsky) [reply src/components/charts/new_revocations/RevocationsByDistrict/RatesChart.js:123]: Done

File: src/components/charts/new_revocations/RevocationsByDistrict/RatesChart.js (hunk 3 above)
Comment (DmitryJakubovsky) [reply src/components/charts/new_revocations/RevocationsByDistrict/RatesChart.js:95]: Done

**Comment by DmitryJakubovsky** [comment]
Hi, @jessex! Can you increase limit for uploaded files, please? I can't upload updated fake data because files occupy more than 100mb.

**Comment by DmitryJakubovsky** [comment]
I did not expect this so I commented `Done` in all issues.

**Comment by jessex** [comment]
>Hi, @jessex! Can you increase limit for uploaded files, please? I can't upload updated fake data because files occupy more than 100mb.

Hey, @DmitryJakubovsky: I looked into this and unfortunately there does not appear to be any way to increase the file upload size limit. If one of these new fixture files goes over 100MB we're going to just have to shrink it. One immediate idea is that I don't _think_ the state code column in the fixtures files is actually required, so we might just be able to drop that out of the file entirely. We could also try to de-prettify the file to get rid of wasted whitespace/etc. If there was no way to get the file under 100MB without removing content, we could try to selectively remote content, e.g. just pick a handful of districts and remove all rows for those districts.

As a last resort, we could consider something more drastic like [Git Large File Storage](https://docs.github.com/en/github/managing-large-files/versioning-large-files). But that will take a bit of time to set up so it would be better to try something from the above to avoid having to do this right now.

File: .eslintignore [hunk 6]
```diff
@@ -43,12 +43,25 @@ public/
 !src/components/charts/facilities/ReincarcerationRateByStayLength.js
 !src/components/charts/facilities/__tests__/**/*.js
 !src/components/charts/new_revocations/helpers.js
+<<<<<<< HEAD
```
Comment (jessex) [inline .eslintignore:46]: Looks we have some unresolved merge conflicts committed in this file.

File: src/components/charts/new_revocations/Chip.scss [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+  &.shrinking {
+    height: 28px;
+    border-radius: 24px;
+
+    .delete {
+      margin-top: 2px;
+    }
+  }
+}
```
Comment (jessex) [inline src/components/charts/new_revocations/Chip.scss:53]: Nit: missing a newline here.

File: src/components/charts/new_revocations/RevocationsByRiskLevel/RevocationsByRiskLevel.js [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+          ],
+        },
+        tooltips: {
+          backgroundColor: COLORS["grey-800-light"],
+          footerFontSize: 9,
+          mode: "index",
+          intersect: false,
+          callbacks: {
+            label: (tooltipItem, data) =>
+              tooltipForRateMetricWithCounts(
```
Comment (jessex) [inline src/components/charts/new_revocations/RevocationsByRiskLevel/RevocationsByRiskLevel.js:177]: For some reason, this doesn't appear to be working as expected. The tooltips for this chart in particular no longer have the numerator and denominator in parentheses. Compare the tooltip for this chart with the tooltip for the race or gender charts to see the difference. At first glance, I don't know why the parenthetical portion is missing here but we need to add it back.

File: src/components/charts/new_revocations/RevocationsByRace/helpers.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+      return "Percent of standing population revoked";
+    case "exits":
+      return "Percent revoked out of all exits";
+  }
+};
+
+/**
+ * Transform to
+ * {
+ *   FEMALE: { LOW: [1, 4], HIGH: [5, 9], ... } }
```
Comment (jessex) [inline src/components/charts/new_revocations/RevocationsByRace/helpers.js:88]: Nit: this docstring copied from the other helpers class is using the wrong keys here.

File: src/components/charts/new_revocations/RevocationsByGender/helpers.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import getOr from "lodash/fp/getOr";
+import pipe from "lodash/fp/pipe";
+import set from "lodash/fp/set";
+import toInteger from "lodash/fp/toInteger";
+import { calculateRate } from "../helpers/rate";
+
+const GENDERS = ["FEMALE", "MALE"];
```
Comment (jessex) [inline src/components/charts/new_revocations/RevocationsByGender/helpers.js:24]: Let's add a comment above here that reads, "These are the only genders that are apparent in the source data set, not all of the genders we would like to represent."

**Pull request merged**